### PR TITLE
Working on clean-up methods to improve surface meshing.

### PR DESCRIPTION
**Pull request by imikejackson**
First, I broke out the determination of node and triangles and the creation of
nodes and triangles from the execute into separate functions.  Next, I added a
function to remove problematic voxel arrangements that will cause self-intersection
of the surface mesh.  Last, I am working on trying to identify and remove loops
in the triple line network - this is not done yet.

These changes were coded by M. Groeber and committed by M. Jackson

Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>

### CODE REVIEW

**Comment by imikejackson** [comment]
These changes were originally written by Mike Groeber. There were some issues with that branch so we moved the code to my fork and submitted the PR.

File: Source/Plugins/SurfaceMeshing/SurfaceMeshingFilters/QuickSurfaceMesh.cpp [hunk 1]
```diff
@@ -368,50 +364,288 @@ void QuickSurfaceMesh::getGridCoordinates(IGeometryGrid::Pointer grid, size_t x,
...
 
   size_t udims[3] = {0, 0, 0};
   std::tie(udims[0], udims[1], udims[2]) = grid->getDimensions();
 
   int64_t dims[3] = {
-      static_cast<int64_t>(udims[0]), static_cast<int64_t>(udims[1]), static_cast<int64_t>(udims[2]),
+      static_cast<int64_t>(udims[0]),
+      static_cast<int64_t>(udims[1]),
+      static_cast<int64_t>(udims[2]),
   };
```
Comment (mmarineBlueQuartz) [inline Source/Plugins/SurfaceMeshing/SurfaceMeshingFilters/QuickSurfaceMesh.cpp:466]: There shouldn't be a comma after the last item.

**Review by mmarineBlueQuartz — APPROVED** [review]
Without going in depth into the algorithm, the code looks good.

File: Source/Plugins/SurfaceMeshing/SurfaceMeshingFilters/QuickSurfaceMesh.cpp (hunk 1 above)
Comment (imikejackson) [reply Source/Plugins/SurfaceMeshing/SurfaceMeshingFilters/QuickSurfaceMesh.cpp:466]: Done.

**Pull request merged**